projects/igniteui-angular/src/lib/time-picker/time-picker.component.ts

```diff
@@ -1145,21 +1162,23 @@ export class IgxTimePickerComponent implements
     public openDialog(timePicker: IgxTimePickerComponent = this): void {
         if (this.mode === InteractionMode.Dialog) {
             this.collapsed = false;
```

Comment:
The check for collapse state can be moved at the beginning of the method and to be performed for both modes.